RevenueCatUI/CustomerCenter/ViewModels/CustomerCenterViewModel.swift

```diff
@@ -43,50 +48,66 @@ import RevenueCat
         }
     }
     @Published
-    var configuration: CustomerCenterConfigData?
+    var configuration: CustomerCenterConfigData? {
+        didSet {
+            // We fail open.
+            let defaultAppIsLatestVersion = true
+
+            guard let currentVersionString = currentAppVersion?.takeVersion() else {
+                self.appIsLatestVersion = defaultAppIsLatestVersion
+                return
+            }
+            guard let latestVersionString = configuration?.lastPublishedAppVersion.takeVersion() else {
+                self.appIsLatestVersion = defaultAppIsLatestVersion
+                return
+            }
+
+            do {
+                let currentVersion = try SemanticVersion(currentVersionString)
+                let latestVersion = try SemanticVersion(latestVersionString)
+                self.appIsLatestVersion = currentVersion >= latestVersion
+            } catch {
+                self.appIsLatestVersion = defaultAppIsLatestVersion
+            }
+      }
+    }
 
     var isLoaded: Bool {
         return state != .notLoaded && configuration != nil
     }
 
     private var customerInfoFetcher: CustomerInfoFetcher
+    private let currentVersionFetcher: CurrentVersionFetcher
     internal let customerCenterActionHandler: CustomerCenterActionHandler?
 
     private var error: Error?
 
-    convenience init(customerCenterActionHandler: CustomerCenterActionHandler?) {
-        self.init(customerCenterActionHandler: customerCenterActionHandler,
-                  customerInfoFetcher: {
-            guard Purchases.isConfigured else {
-                throw PaywallError.purchasesNotConfigured
-            }
-
+    init(
+        customerCenterActionHandler: CustomerCenterActionHandler?,
+        customerInfoFetcher: @escaping CustomerInfoFetcher = {
+            guard Purchases.isConfigured else { throw PaywallError.purchasesNotConfigured }
             return try await Purchases.shared.customerInfo()
-        })
-    }
-
-    init(customerCenterActionHandler: CustomerCenterActionHandler?,
-         customerInfoFetcher: @escaping CustomerInfoFetcher) {
+        },
+        currentVersionFetcher: @escaping CurrentVersionFetcher = {
+            Bundle.main.infoDictionary?["CFBundleVersion"] as? String
```

Comment:
These have been confusing me haha. 

[The docs for `CFBundleVersion`](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion) say: 
>This key is a machine-readable string composed of one to three period-separated integers, such as 10.14.1. The string can only contain numeric characters (0-9) and periods.

Which appears to be exactly what we need? 

Then again, [the docs for `CFBundleShortVersionString`](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring) say: 
>This key is a user-visible string for the version of the bundle. The required format is three period-separated integers, such as 10.14.1. The string can only contain numeric characters (0-9) and periods.

So 🤷 

I had initially chosen `CFBundleVersion` because the docs claim it's intended to be machine readable.

Reply:
It is confusing, and the docs are... wrong. Give it a test! Also note our existing code: https://github.com/RevenueCat/purchases-ios/blob/9eec64d82ec3052d07979f012115f80c024d8b3e/Sources/Misc/SystemInfo.swift#L85-L90

Reply:
I also have some recollection of this behaving stupidly different in macOS vs native iOS for some reason, this stuff is beyond confusing

Reply:
That rings a bell...

Reply:
Did not consider docs being wrong. 😅 Thanks for being vigilant! 
 
Would be nice to use the existing `SystemInfo`, but somehow that is not accessible in the scope. Not entirely sure why. I'm using `CFBundleShortVersionString` directly now.

Reply:
Ah right, of course. Oh well, this does the trick.